x-pack/plugins/ml/public/data_frame/components/source_index_preview/source_index_preview.tsx

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useContext, useEffect, useState } from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { SearchResponse } from 'elasticsearch';
+
+import {
+  EuiButtonEmpty,
+  EuiButtonIcon,
+  EuiCheckbox,
+  EuiEmptyPrompt,
+  EuiFlexGroup,
+  EuiFlexItem,
+  // @ts-ignore
+  EuiInMemoryTable,
+  EuiPanel,
+  EuiPopover,
+  EuiPopoverTitle,
+  EuiProgress,
+  EuiText,
+  EuiTitle,
+  RIGHT_ALIGNMENT,
+} from '@elastic/eui';
+
+import { ml } from '../../../services/ml_api_service';
+
+import { Dictionary } from '../../../../common/types/common';
+
+import { IndexPatternContext, SimpleQuery } from '../../common';
+
+import {
+  EsDoc,
+  EsFieldName,
+  getDefaultSelectableFields,
+  getSelectableFields,
+  MAX_COLUMNS,
+  toggleSelectedField,
+} from './common';
+import { ExpandedRow } from './expanded_row';
+
+type ItemIdToExpandedRowMap = Dictionary<JSX.Element>;
+
+interface Sorting {
+  sort: {
+    field: string;
+    direction: 'asc' | 'desc';
```

Comment:
These sorting directions could probably live in a general enum file? Or is there already an EUI enum for them?

Reply:
I tried to use EUI's `SortDirection` but wasn't able to use it with this union type. So for this file I created another enum in 6b4b1f4.